File: OOP-Java/Java Basics/private_constructor_demo/Car.java
/*
   Object creation of a class having private constructor
 */

class Vehicle {

    private Vehicle(){
        System.out.println("constructor of Vehicle");
    }

    public static Vehicle getInstance() {
        return new Vehicle();
    }
}

public class Car {
    public static void main(String[] args) {
        Vehicle v = Vehicle.getInstance();
    }
}

/*
   constructor of Vehicle
 */

File: OOP-C++/Part3/Output/Output1/Car.cpp
class Car : public Vehicle {

	public :
		int numGears;

	Car() : Vehicle(5) {
		cout << "Car's default constructor " << endl;
	}

	~Car() {
		cout << "Car's Destructor " << endl;
	}

	void print() {
		cout << "NumTyres : " << numTyres << endl;
		cout << "Color : " << color << endl;
		cout << "Num gears :" << numGears << endl;
	}
};
